dll/win32/imm32/precomp.h

```diff
@@ -93,7 +93,14 @@ static inline BOOL Imm32IsCrossProcessAccess(HWND hWnd)
             (DWORD_PTR)NtCurrentTeb()->ClientId.UniqueProcess);
 }
 
-#define ImeDpi_IsUnicode(pImeDpi) ((pImeDpi)->ImeInfo.fdwProperty & IME_PROP_UNICODE)
+#define ImeDpi_IsUnicode(pImeDpi)   ((pImeDpi)->ImeInfo.fdwProperty & IME_PROP_UNICODE)
+#define Imm32IsImmMode()            (g_psi && (g_psi->dwSRVIFlags & SRVINFO_IMM32))
+#define Imm32IsCiceroMode()         (g_psi && (g_psi->dwSRVIFlags & SRVINFO_CICERO_ENABLED))
+
+static inline BOOL Imm32IsSomethingMode(VOID) /* TODO: unknown */
+{
+    return !!(((PW32CLIENTINFO)NtCurrentTeb()->Win32ClientInfo)->W32ClientInfo[0] & 2);
```

Comment:
This is the check of `TIF_16BIT` in the thread's `THREADINFO::TIF_flags`
See https://reactos.org/wiki/Techwiki:Win32k/THREADINFO and https://www.geoffchappell.com/studies/windows/km/win32k/structs/threadinfo/tif_flags.htm
and our user32.

Reply:
I had to move some structures from `win32.h` to `ntuser.h`...

Reply:
Why? user32 code can access them as they currently are 🤔

Reply:
1. We cannot include `win32.h`.